pyproject.toml

```diff
@@ -31,6 +31,7 @@ requests = "^2.31.0"
 llama-cpp-python = "^0.2.56,!=0.2.86"
 drain3 = "^0.9.11"
 huggingface-hub = ">0.23.2"
+numpy = "^2.1.0"
```

Comment:
Fedora 40 has numpy 1.26.4. Both `median` and `exp` are present there. So `>= 1.26.4' is ok. The API interface changed last time in 2008 (about 1.2 release). `exp` was hard to find. But I believe it is even older. Therefore I recommend removing any version requirement and require simply `numpy`.